RoddyCore/src/de/dkfz/roddy/Roddy.java

```diff
@@ -399,19 +399,39 @@ private static void initializeFeatureToggles() {
             });
         }
         //Override toggles in ini
-        if (commandLineCall.isOptionSet(RoddyStartupOptions.enabletoggles)) {
-            List<String> enabledToggles = commandLineCall.getOptionList(RoddyStartupOptions.enabletoggles);
-            enabledToggles.forEach(toggle -> {
-                if (AvailableFeatureToggles.valueOf(toggle) != null) featureToggleConfig.setProperty(toggle, "true");
-            });
+        boolean successful = true;
+        successful &= retrieveAndSetToggleValuesFromCLI(RoddyStartupOptions.enabletoggles, true);
+        successful &= retrieveAndSetToggleValuesFromCLI(RoddyStartupOptions.disabletoggles, false);
+
+        // TODO:STRICT In Strict mode we should exit after all toggles were checked.
+        if(!successful) {
+            String toggleNames = RoddyIOHelperMethods.joinArray(AvailableFeatureToggles.values(), "\n\t");
+            logger.severe("Available toggle values are:\n\t" + toggleNames);
         }
+    }
 
-        if (commandLineCall.isOptionSet(RoddyStartupOptions.disabletoggles)) {
-            List<String> enabledToggles = commandLineCall.getOptionList(RoddyStartupOptions.enabletoggles);
-            enabledToggles.forEach(toggle -> {
-                if (AvailableFeatureToggles.valueOf(toggle) != null) featureToggleConfig.setProperty(toggle, "false");
-            });
+    private static boolean retrieveAndSetToggleValuesFromCLI(RoddyStartupOptions opt, final boolean enabled) {
+        if (!(opt == RoddyStartupOptions.enabletoggles || opt == RoddyStartupOptions.disabletoggles)) {
+            // If the method is misused, we just exit. This is a programmatic error and should not punish the user.
+            return true;
         }
+
+        // Stores if an error happened
+        boolean error = false;
+        if (commandLineCall.isOptionSet(opt)) {
+            List<String> listOfToggles = commandLineCall.getOptionList(opt);
+            for(String toggle : listOfToggles) {
+                try {
+                    if (AvailableFeatureToggles.valueOf(toggle) != null)
```

Comment:
Yes you're right, but that is exactly what is happening three lines down.
Actually the query throws an exception but I did not come up with a short way to make a safe version in Java.

However, if you look into the catch part, there is the error message and the toggle name.
Also, if the method returns false, a list of the available toggles will be printed.